module2/homework/nwdNww/nwdNww.hpp

```diff
@@ -1,11 +1,29 @@
 #pragma once
+#include <cmath>
 
 int NWD(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+    int mod;
+
+    if(lhs < 0) {
+        lhs = -lhs;
+    }
+    if(rhs < 0) {
+        rhs = -rhs;
+    }
+
+    while(rhs) {
+        mod = lhs % rhs;
+        lhs = rhs;
+        rhs = mod;
+    }
+    return lhs;
 }
 
 int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+    int nwd = NWD(lhs, rhs);
+    if(!nwd) {
+        return 0;
+    }
+
+    return abs(lhs * rhs) / nwd;
```

Comment:
You don't even need `nwd` variable. NWD is only equals 0 if min. one of the input variables equals 0.
```suggestion
    if(lhs == 0 || rhs == 0) {
        return 0;
    }

    return abs(lhs * rhs) / NWD(lhs, rhs);
```

Reply:
I'm copying my comment from different PR:

There are trivial cases that your code won't work for, ie NWW(1e8, 1e8).
That is due to lhs*rhs overflowing int.
You can fix it by casting lhs to unsigned long long in the multiplication.

There are some nontrivial cases that will still fail, ie NWW(1e8-1, 1e8+1), but they fail for the std:lcm(int, int) as well, so it's ok.